portal-backend/depmap/compute/views.py

```diff
@@ -161,14 +161,28 @@ def post(self):
                 query_values=query_values,
             )
 
+        dataset_depmap_ids = set(data_access.get_dataset_sample_ids(dataset_id))
+
         # Two-class comparison case
         if analysis_type == AnalysisType.two_class:
-            cl_query_vector = query_cell_lines
+            cl_query_vector = query_cell_lines # includes both in group and out group
             assert all(
                 x in {"in", "out"} for x in query_values
             ), "Expecting values in {} to be either 'in' or 'out'".format(query_values)
             value_query_vector = [0 if x == "out" else 1 for x in query_values]
 
+            # Validate that BOTH the in-group and out-group have cell lines present in the dataset
+            in_group_cell_lines = {query_cell_lines[i] for i in range(len(query_values)) if query_values[i] == "in"}
+            out_group_cell_lines = {query_cell_lines[i] for i in range(len(query_values)) if query_values[i] == "out"}
+            if len(in_group_cell_lines.intersection(dataset_depmap_ids)) == 0:
+                return format_taskless_error_message(
+                    "No cell lines in common between in-group and dataset selected"
+                )
+            if len(out_group_cell_lines.intersection(dataset_depmap_ids)) == 0:
+                return format_taskless_error_message(
+                    "No cell lines in common between out-group and dataset selected"
+                )
```

Comment:
I don't like duplicating this fix, but there's already a lot of duplicate custom analysis code between breadbox and the portal-backend. Once we've fully migrated our data to breadbox, I'm hoping we can just get rid of the portal-backend implementation entirely.